manifests/init.pp

```diff
@@ -64,4 +68,8 @@
     rulesets      => $config_rulesets,
     templates     => $config_templates,
   }
+
+  class { 'systemd':
```

Comment:
This seems a bit odd.

Wouldn't we expect other modules might include the base `systemd` module/class (in the future)?

Is there a way to specifically only set the journald settings after making sure `systemd` module is included?

Reply:
That sounds reasonable. But I haven't thought it through very extensively.

But I don't think you can specify hiera defaults in this module for `systemd::journald_settings`.
There is a private class `systemd::journald` but I'm guessing you can't call that here.

If this is too painful, we can just live with calling the `systemd` class here as you have it and address it later if/when it becomes an issue.

Reply:
Yep, declaring the `systemd::journald` class on its own doesn't work (as you suspected, because it's private; plus it doesn't have any parameters of its own, which defeats the purpose of declaring it rather than just including it).

And also yep, if I add `systemd::journald_settings` into module Hiera for `profile_rsyslog` and just `include systemd` in rsyslog, those settings are ignored. I think just because the systemd data is "foreign" to the profile_rsyslog module.

But...we can just put our defaults into `control-repo` and our other control repos. It's easier to rework it this way now rather than refactor later. Admins using our documented process for pulling in updates from control-repo should see the changes in `control-repo`.

I'll code up the change in a new commit.

Reply:
2nd commit has the changes. Ready for another look when you get a chance. Thanks!

Reply:
I think we need to add a note in the README.md that you need to set the `systemd::journald_settings` parameters in your control repo.

Reply:
Good idea! https://github.com/ncsa/puppet-profile_rsyslog/tree/rundall/SVCPLAN-3421/disable_systemd_journald_rate_limiting#mangement-of-systemd-journald